src/shared/common/ScaleBar.js

```diff
@@ -28,100 +32,105 @@ const getScaleBar = (meters, feet, tileWidth, referenceSize) => {
     const top = 0;
     const mid = 16;
     // convert meters and feet into "pixels" so that we draw at the correct scale!
-    const metersPx = (meters / tileWidth) * referenceSize;
-    const feetPx = (feet / tileWidth) * referenceSize;
+    const metersPx = (meters / tileWidthInMeters) * referenceSize;
+    const feetPx = (feet / tileWidthInMeters) * referenceSize;
     const bottom = top + 2 * (mid - top);
-    const p = Path().moveTo(0, top);
-    p.lineTo(0, bottom);
-    p.moveTo(0, mid);
-    p.lineTo(metersPx, mid);
-    p.lineTo(metersPx, top);
-    p.moveTo(metersPx, mid);
-    p.lineTo(feetPx * 0.3048, mid);
-    p.lineTo(feetPx * 0.3048, bottom);
+    const parts = [
+        `M0 ${top}`,
+        `L0 ${bottom}`,
+        `M0 ${mid}`,
+        `L${metersPx} ${mid}`,
+        `L${metersPx} ${top}`,
+        `M${metersPx} ${mid}`,
+        `L${feetPx * 0.3048} ${mid}`,
+        `L${feetPx * 0.3048} ${bottom}`,
+    ];
+    const p = parts.join(' ');
     return p;
 };
 
 export default (props: Props): React.Node => {
-    const { alignToBottom, latitude, useScreenWidth, visible, zoomLevel } =
-        props;
+    const {
+        alignToBottom,
+        latitude,
+        position,
+        referenceSize,
+        // useScreenWidth,
+        visible,
+        zoomLevel,
+    } = props;
 
     // calculate the width of one tile (in meters)
     // this magic formula comes from
     // https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#Resolution_and_Scale
     // This assumes that each image is 256 pixels wide, which may not be
     // the case for specific providers. Adjustments might be needed if
     // this case arises.
-    const tileWidth =
+    const tileWidthInMeters =
         (Math.cos(latitude * (Math.PI / 180)) * 2 * Math.PI * 6378137) /
         2 ** zoomLevel;
     let feet;
     let meters;
-    // we hardcode the scale bar sizes, and pick an appropriate one
-    // for the current zoom level
-    switch (true) {
-        case tileWidth < 70:
-            meters = 30;
-            feet = 100;
-            break;
-        case tileWidth < 110:
-            meters = 50;
-            feet = 200;
-            break;
-        case tileWidth < 180:
-            meters = 100;
-            feet = 300;
-            break;
-        default:
-            meters = 200;
-            feet = 500;
-            break;
+    // calculate scalebar size so that it fits in roughly half the image
+    // display width, while using a rough step function. This may give
+    // somewhat strange results in high latitudes where the rounding
+    // will not look nice.
+    if (tileWidthInMeters < 200) {
+        meters = Math.trunc(tileWidthInMeters / 10 / 2) * 10;
+        feet = Math.round(meters / 0.3048 / 100) * 100;
```

Comment:
mostly because the shortest scalebar I've seen was 30m/100ft, so the rounding to smaller values isn't really used. I feel like it's easier to picture 100m/300ft than 100m/310ft?